test/e2e/self_hosted_test.go

```diff
@@ -22,103 +22,103 @@ import (
 )
 
 var _ = Describe("Self-hosted", Serial, func() {
-	provider := "Docker"
-	lowercaseProvider := strings.ToLower(provider)
-	for _, cniProvider := range []string{"Cilium", "Calico"} {
-		for _, addonStrategy := range []string{"HelmAddon", "ClusterResourceSet"} {
-			strategy := ""
-			switch addonStrategy {
-			case "HelmAddon":
-				strategy = "helm-addon"
-			case "ClusterResourceSet":
-				strategy = "crs"
-			default:
-				Fail("unknown addon strategy: " + addonStrategy)
-			}
-			flavour := fmt.Sprintf(
-				"topology-%s-%s",
-				strings.ToLower(cniProvider),
-				strategy,
-			)
-			Context(
-				flavour,
-				Label("provider:"+provider),
-				Label("cni:"+cniProvider),
-				Label("addonStrategy:"+addonStrategy),
-				func() {
-					framework.SelfHostedSpec(ctx, func() framework.SelfHostedSpecInput {
-						return framework.SelfHostedSpecInput{
-							E2EConfig:              e2eConfig,
-							ClusterctlConfigPath:   clusterctlConfigPath,
-							BootstrapClusterProxy:  bootstrapClusterProxy,
-							ArtifactFolder:         artifactFolder,
-							SkipCleanup:            skipCleanup,
-							Flavor:                 flavour,
-							InfrastructureProvider: ptr.To(lowercaseProvider),
-							PostClusterMoved: func(proxy capiframework.ClusterProxy, cluster *clusterv1.Cluster) {
-								By(
-									"Waiting for all requested addons to be ready in workload cluster",
-								)
-								workloadCluster := capiframework.GetClusterByName(
-									ctx,
-									capiframework.GetClusterByNameInput{
-										Namespace: cluster.GetNamespace(),
-										Name:      cluster.GetName(),
-										Getter:    proxy.GetClient(),
-									},
-								)
-								Expect(workloadCluster.Spec.Topology).ToNot(BeNil())
-								clusterVars := variables.ClusterVariablesToVariablesMap(
-									workloadCluster.Spec.Topology.Variables,
-								)
-								addonsConfig, err := variables.Get[apivariables.Addons](
-									clusterVars,
-									v1alpha1.ClusterConfigVariableName,
-									"addons",
-								)
-								Expect(err).ToNot(HaveOccurred())
-								WaitForAddonsToBeReadyInWorkloadCluster(
-									ctx,
-									WaitForAddonsToBeReadyInWorkloadClusterInput{
-										AddonsConfig:           addonsConfig,
-										ClusterProxy:           proxy,
-										WorkloadCluster:        workloadCluster,
-										InfrastructureProvider: lowercaseProvider,
-										DeploymentIntervals: e2eConfig.GetIntervals(
-											flavour,
-											"wait-deployment",
-										),
-										DaemonSetIntervals: e2eConfig.GetIntervals(
-											flavour,
-											"wait-daemonset",
-										),
-										HelmReleaseIntervals: e2eConfig.GetIntervals(
-											flavour,
-											"wait-helmrelease",
-										),
-										ClusterResourceSetIntervals: e2eConfig.GetIntervals(
-											flavour,
-											"wait-clusterresourceset",
-										),
-									},
-								)
+	for _, provider := range []string{"Docker", "Nutanix"} {
```

Comment:
The diff is hard to read, any other changes besides adding the provider?

Reply:
Yes just adding the provider here.

Much easier if you switch to side by side and ignore whitespace btw...

<img width="1352" alt="image" src="https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/assets/464659/c6ee363b-a07d-4688-9697-25e539161557">